src/base/oven.lua

```diff
@@ -538,6 +538,9 @@
 		local system = os.target()
 		local architecture = os.targetarch()
 		local toolset = p.action.current().toolset
+		if type(toolset) == "function" then
+			toolset = toolset()
```

Comment:
Why does this need to be a function and not evaluated in place as it used to be?